src/models/test/ThemeProvider.test.js

```diff
@@ -0,0 +1,117 @@
+/* eslint-disable react/no-multi-comp */
+import React from 'react'
+import expect from 'expect'
+import { shallow, render } from 'enzyme'
+import ThemeProvider, { CHANNEL } from '../ThemeProvider'
+
+describe('ThemeProvider', () => {
+  it('should not throw an error when no children are passed', () => {
+    const result = shallow(<ThemeProvider theme={{}} />)
+    expect(result.node).toEqual(null)
+  })
+
+  it('should accept a theme prop that\'s a plain object', () => {
+    shallow(<ThemeProvider theme={{ main: 'black' }} />)
+  })
+
+  it('should render its child', () => {
+    const child = (<p>Child!</p>)
+    const renderedComp = shallow(
+      <ThemeProvider theme={{ main: 'black' }}>
+        { child }
+      </ThemeProvider>
+    )
+    expect(renderedComp.contains(child)).toEqual(true)
+  })
+
+  it('should merge it\'s theme with an outer theme', (done) => {
+    const outerTheme = { main: 'black' }
+    const innerTheme = { secondary: 'black' }
+    // Setup Child
+    class Child extends React.Component {
+      componentWillMount() {
+        this.context.broadcasts[CHANNEL](theme => {
+          expect(theme).toEqual(Object.assign({}, outerTheme, innerTheme))
```

Comment:
bit confused about this here. Why isn't the assertion on the result of the `render`?